saga/sagaCoordinator.go

```diff
@@ -0,0 +1,79 @@
+package saga
+
+//
+// Saga Object which provides all Saga Functionality
+// Implementations of SagaLog should provide a factory method
+// which returns a saga based on its implementation.
+//
+type SagaCoordinator struct {
+	log SagaLog
+}
+
+//
+// Make a Saga which uses the specied SagaLog interface for durable storage
+//
+func MakeSagaCoordinator(log SagaLog) SagaCoordinator {
+	return SagaCoordinator{
+		log: log,
+	}
+}
+
+// Make a Saga add it to the SagaCoordinator, if a Saga Already exists
+// with the same id, it will overwrite the already existing one.
```

Comment:
Oh sorry this comment is actually out of date.  I was originally exploring the idea of keeping a map of all the inprogress Sagas in the SagaCoordinator.  But that made the code more complicated and I couldn't think of a good reason why the SagaCoordinator needs to keep track of each running Saga right now (obviously we can add this later)  So the comment is form that context.  Fixed comment

What will actually happen is each Saga object will have its own channel to do updates. The SagaLog will have multiple entries

StartSaga
StartTask1
StartSaga
StartTask1
EndTask1
EndSaga
EndTask1
EndSaga

But that's ok, Recovery and reading state from the Saga Log handles this and will handle this correctly.  We can have this scenario if two schedulers run the same job (Aurora can sometimes have ghost instances during leader election).